consensus concepts

ElPaisano opened the pull request:
Adds concepts stuff about expected consensus. Ended up modifying the tipsets (which is important in understanding EC) page, as well as making adjustments to the proofs page (which is also referenced in the EC page)

Addresses https://github.com/filecoin-project/filecoin-docs/issues/1803 and https://github.com/filecoin-project/filecoin-docs/issues/1830

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -1,7 +1,7 @@
 title: "Consensus"
 description: ""
-lead: ""
+lead: "In the Filecoin blockchain, network consensus is achieved using the Expected Consenus (EC) algorithm, a secret, fair and verifiable consensus protocol used by the network to agree on the overall chain state at each epoch, including the storage power currently available in the network."
```
I'm not sure Expected Consensus needs to be capitalized.

ElPaisano replied:
According to https://github.com/filecoin-project/consensus and https://spec.filecoin.io/algorithms/expected_consensus/, it looks like it should be capitalized

johnnymatthews replied:
Ah, ok then. Leave it as it is.

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -11,5 +11,49 @@ menu:
+
+In the Filecoin blockchain, network _{{< tooltip "consensus" >}}_ is achieved using the Expected Consenus (EC) algorithm, a probabilistic, _{{< tooltip "Byzantine fault-tolerant" >}}_ consensus protocol. At a high-level, EC achieves consensus by running a secret, fair and verifiable _leader election_ at every {{< tooltip "epoch" >}} where a set number of participants may become eligible to submit ...
+
+## Properties of EC
```
It's not necessary to callout _X of EC_ in each title. The reader knows we're talking about EC because we're on the _EC_ page.

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -11,5 +11,49 @@ menu:
+- A block producer can be verified by any participant in the network.
+- The identity of a block producer is anonymous until they release their block to the network.
+
+## Steps in EC
```
It's not necessary to callout _X of EC_ in each title. The reader knows we're talking about EC because we're on the _EC_ page.

ElPaisano replied:
Removed in https://github.com/filecoin-project/filecoin-docs/pull/1831/commits/c0e4b87c18f60f33bfc86ff7194cb9c908690e80

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -11,5 +11,49 @@ menu:
+
+- Each epoch has potentially multiple elected leaders who may propose a block.
+- A winner is elected randomly from a set of network participants weighted according to the respective storage power they contribute to the Filecoin network. 
+- All blocks proposed are grouped together in a _tipset_ (discussed below), from which the final chain is selected.
```
```suggestion
- All blocks proposed are grouped together in a _tipset_, from which the final chain is selected.
```

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -11,5 +11,49 @@ menu:
+1. Each potential block producer elected generates a storage proof using [WinningPoSt]({{< relref "proofs.md#winningpost" >}}) for a randomly selected _{{< tooltip "sector" >}}_ within in short window of time. Potential block producers that fail this step are not eligible to produce a block. In this step, the following could occur:
+
+   - **All potential block producers fail WinningPoSt**, in which case EC returns to step 1 (described above).
+   - **One or more potential block producers passes WinningPoSt**, which means they are eligible to submit that block to the epochs tipset (discussed in the next step)
```
```suggestion
   - **One or more potential block producers passes WinningPoSt**, which means they are eligible to submit that block to the epochs tipset.
```

johnnymatthews commented on content/en/basics/the-blockchain/consensus/index.md:
```diff
@@ -11,5 +11,49 @@ menu:
+
+In summary, EC involves the following steps at each _{{< tooltip "epoch" >}}_:
+
+1. A storage provider checks to see if they are elected to propose a block by generating an [_election proof_](https://spec.filecoin.io/#section-glossary.election-proof).
+
+1. Zero, one or multiple storage providers may be elected to propose a block. This does not mean that an elected participant is guaranteed to be able to submit a block. In the case where:
+
+   - **No storage providers are elected to propose a block in a given epoch**, a new election is run in the next epoch to ensure that the network remains live.
+   - **One or more storage providers are elected to propose a block in a given epoch**, each must generate a _WinningPoSt proof-of-storage_ to be eligible to actually submit a block.
```
```suggestion
1. A storage provider checks to see if they are elected to propose a block by generating an [_election proof_](https://spec.filecoin.io/#section-glossary.election-proof).
1. Zero, one or multiple storage providers may be elected to propose a block. This does not mean that an elected participant is guaranteed to be able to submit a block. In the case where:

   - **No storage providers are elected to propose a block in a given epoch**, a new election is run in the next epoch to ensure that the network remains live.
   - **One or more storage providers are elected to propose a block in a given epoch**, each must generate a _WinningPoSt proof-of-storage_ to be eligible to actually submit a block.
```

Don't leave an empty space between list items.

johnnymatthews reviewed (changes requested):
This is a great PR! Just left a few minor change requests:

- Don't capitalize Expected Consensus
- Don't leave an empty space between list items.
- Try not to use parenthesis.

ElPaisano commented:
> This is a great PR! Just left a few minor change requests:
> 
> * Don't capitalize Expected Consensus
> * Don't leave an empty space between list items.
> * Try not to use parenthesis.

@johnnymatthews  Thanks for the review! Addresses all points expect capitalization of EC - see https://github.com/filecoin-project/filecoin-docs/pull/1831#discussion_r1143669662
